File: CS/Serialization/MyDeserializeHelper.cs
using DevExpress.Utils.Serializing.Helpers;

namespace Serialization
{
	public class MyDeserializeHelper : DeserializeHelper
	{
		public MyDeserializeHelper(object rootObject)
			: base(rootObject)
		{
		}
		public MyDeserializeHelper(object rootObject, bool resetProperties)
			: base(rootObject, resetProperties)
		{
		}
		public MyDeserializeHelper()
			: base()
		{
		}

		protected override SerializationContext CreateSerializationContext()
		{
			MySerializationContext newContext = new MySerializationContext();
			newContext.DeserializeHelper = this;

			return newContext;
		}

		internal object RootObject
		{
			get
			{
				return rootObject;
			}
		}
	}
}

File: VB/Serialization/MyDeserializeHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Utils.Serializing.Helpers

Namespace Serialization
	Public Class MyDeserializeHelper
		Inherits DeserializeHelper
		Public Sub New(ByVal rootObject As Object)
			MyBase.New(rootObject)
		End Sub
		Public Sub New(ByVal rootObject As Object, ByVal resetProperties As Boolean)
			MyBase.New(rootObject, resetProperties)
		End Sub
		Public Sub New()
			MyBase.New()
		End Sub

		Protected Overrides Function CreateSerializationContext() As SerializationContext
			Dim newContext As New MySerializationContext()
			newContext.DeserializeHelper = Me

			Return newContext
		End Function

		Friend ReadOnly Overloads Property RootObject() As Object
			Get
				Return rootObject
			End Get
		End Property
	End Class
End Namespace
